docs/containers-transports.5.md

```diff
@@ -0,0 +1,53 @@
+% CONTAINERS-TRANSPORTS(5) Containers Transports Man Page
+% Valentin Rothberg
+% April 2019
+
+# NAME
+containers-transports - description of supported transports for copying container images
+
+# DESCRIPTION
+
+**containers-storage:**_docker-reference_
+
+An image located in a local containers storage image store.
+Please refer to containers-storage.conf(5) for further information.
+
+**dir:**_path_
+
+An existing local directory _path_ storing the manifest, layer tarballs and signatures as individual files.
+This is a non-standardized format, primarily useful for debugging or noninvasive container inspection.
+
+**docker://**_docker-reference_
+
+An image in a registry implementing the "Docker Registry HTTP API V2".
+By default, uses the authorization state in either `$XDG_RUNTIME_DIR/containers/auth.json`, which is set using `(podman login)`.
+If the authorization state is not found there, `$HOME/.docker/config.json` is checked, which is set using `(docker login)`.
+
+**docker-archive:**_path_[**:**_docker-reference_]
+
+An image is stored in the `docker save` formatted file.
+_docker-reference_ is only used when creating such a file, and it must not contain a digest.
+Note that it is possible to copy data to stdin by specifying `docker-archive:/dev/stdin`.
+
+**docker-daemon:**_docker-reference_
+
+An image _docker-reference_ stored in the docker daemon internal storage.
+_docker-reference_ must contain either a tag or a digest.
```

Comment:
digests are only acceptable when reading images, not when used as a destination.

Reply:
```
FATA[0000] Error determining manifest MIME type for docker-daemon:alpine@sha256:28ef97b8686a0b5399129e9b763d5b7e5ff03576aa5580d6f4182a49c5fe1913: Manifest does not match provided manifest digest sha256:28ef97b8686a0b5399129e9b763d5b7e5ff03576aa5580d6f4182a49c5fe1913
```

Using the digest as source does not work atm (Cc @nalind).

Reply:
Oops, I never thought of that; and the daemon does not record the original manifest (and can’t provide the original compressed blobs), so this can’t work unless the transport could just opt out of digest verification (or lie and say that the reference did not contain a digest).

Anyway, the point was primarily that _destinations_ must not use digests.